src/pocketmine/Player.php

```diff
@@ -1184,6 +1184,7 @@ public function setSpawn(Vector3 $pos){
 		$pk->y = (int) $this->spawnPosition->y;
 		$pk->z = (int) $this->spawnPosition->z;
 		$pk->spawnType = SetSpawnPositionPacket::TYPE_PLAYER_SPAWN;
+		$pk->spawnForced = true;
```

Comment:
This pull request is primarily fixing a core bug, feature additions should be done separately.